test/e2e/kubectl.go

```diff
@@ -188,6 +195,114 @@ var _ = Describe("Kubectl client", func() {
 			if e, a := "hi", execOutput; e != a {
 				Failf("Unexpected kubectl exec output. Wanted %q, got %q", e, a)
 			}
+
+			By("executing a command in the container through an HTTP proxy")
+			// Start by checking for the environment variable before anything else
+			kubectlBinPath := os.Getenv("KUBECTLBIN")
+			// Fail if the variable isn't set
+			if kubectlBinPath == "" {
+				Failf("KUBECTLBIN environment variable must be set to the full path of the kubectl binary")
+			}
+
+			apiserverAddr := os.Getenv("APISERVERADDR")
+			// Fail if the variable isn't set
+			if apiserverAddr == "" {
+				Failf("APISERVERADDR environment variable must be set to the full address to the api server")
+			}
+
+			// start the proxy container
+			goproxyPodPath := filepath.Join(testContext.RepoRoot, "test/images/goproxy/pod.yaml")
+			runKubectl("create", "-f", goproxyPodPath, fmt.Sprintf("--namespace=%v", ns))
+
+			// start exec-proxy-tester container
+			netexecPodPath := filepath.Join(testContext.RepoRoot, "contrib/for-tests/netexec/pod.yaml")
+			runKubectl("create", "-f", netexecPodPath, fmt.Sprintf("--namespace=%v", ns))
+			checkPodsRunningReady(c, ns, []string{goproxyContainer, netexecContainer}, podStartTimeout)
+
+			// get the proxy address
+			goproxyPod, err := c.Pods(ns).Get(goproxyContainer)
+			if err != nil {
+				Failf("Unable to get the goproxy pod. Error: %s", err)
+			}
+			proxyAddr := fmt.Sprintf("http://%s:8080", goproxyPod.Status.PodIP)
+			// get the exec address
+			netexecPod, err := c.Pods(ns).Get(netexecContainer)
+			netexecAddr := fmt.Sprintf("http://%s:8080", netexecPod.Status.PodIP)
+
+			// upload kubectl to remote exec server
+			postBodyBuffer := &bytes.Buffer{}
```

Comment:
We may want to consider using a pipe to stream the upload instead of copying the entire kubectl binary into memory prior to uploading. See https://github.com/gebi/go-fileupload-example/blob/master/main.go for an example.